tests/utilities/test_auto_restart.py

```diff
@@ -720,7 +723,9 @@ def on_train_batch_start(self, trainer, *_) -> None:
                 assert not isinstance(loaders["a"][2].loader.dataset, CaptureMapDataset)
                 assert isinstance(loaders["b"].loader.dataset, RangeIterableDataset)
 
-    with mock.patch.dict(os.environ, {"PL_FAULT_TOLERANT_TRAINING": use_fault_tolerant}):
+    with mock.patch.dict(os.environ, {"PL_FAULT_TOLERANT_TRAINING": use_fault_tolerant}), mock.patch(
```

Comment:
The lambda is normally not necessary

Reply:
Moved outside of the test. It is necessary or the test would fail as multiple dataset are available.